staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/tls_test.go

```diff
@@ -53,23 +53,25 @@ func TestTLSConnection(t *testing.T) {
 	certFile, keyFile, caFile := configureTLSCerts(t)
 	defer os.RemoveAll(filepath.Dir(certFile))
 
-	tlsInfo := &transport.TLSInfo{
+	// override server config to be TLS-enabled
+	etcdConfig := testserver.NewTestConfig(t)
+	etcdConfig.ClientTLSInfo = transport.TLSInfo{
 		CertFile:      certFile,
 		KeyFile:       keyFile,
 		TrustedCAFile: caFile,
 	}
+	for i := range etcdConfig.LCUrls {
```

Comment:
Nit suggestion: maybe we could allow this config to be configurable in the `NewTestConfig`?

Reply:
there's so much in the config object, I think the plumbing would converge to just the config struct... I'm open to revisiting if we see a lot of callers tweaking this

Reply:
sounds good to me 👍